shared/data/core/src/main/java/com/ivy/data/repository/TransactionRepository.kt

```diff
@@ -97,29 +97,35 @@ class TransactionRepository @Inject constructor(
         accountId: AccountId,
         startDate: Instant,
         endDate: Instant
-    ): List<Transaction> = retrieveTrns(
-        dbCall = {
-            transactionDao.findAllByAccountAndBetween(
-                accountId = accountId.value,
-                startDate = startDate,
-                endDate = endDate
-            )
+    ): List<Transaction> = withContext(dispatchersProvider.io) {
+        val transactions = transactionDao.findAllByAccountAndBetween(
+            accountId = accountId.value,
+            startDate = startDate,
+            endDate = endDate
+        )
+        val tagAssociationMap = getTagsForTransactionIds(transactions)
+        transactions.mapNotNull {
+            val tags = tagAssociationMap[it.id] ?: emptyList()
+            with(mapper) { it.toDomain(tags = tags).getOrNull() }
         }
-    )
+    }
 
     suspend fun findAllToAccountAndBetween(
         toAccountId: AccountId,
         startDate: Instant,
         endDate: Instant
-    ): List<Transaction> = retrieveTrns(
-        dbCall = {
-            transactionDao.findAllToAccountAndBetween(
-                toAccountId = toAccountId.value,
-                startDate = startDate,
-                endDate = endDate
-            )
+    ): List<Transaction> = withContext(dispatchersProvider.io) {
```

Comment:
The `retrieveTrns` method has a parameter for tags. Can we use it like this?
```kotlin
withContext(dispatchersProvider.io) {
            val tagMap = async { findTagsForTransactionIds(ids) }
            retrieveTrns(
                dbCall = {
                    transactionDao.findByIds(ids.map { it.value })
                },
                retrieveTags = {
                    tagMap.await()[it.id] ?: emptyList()
                }
            )
        }
    }

```

Reply:
But for that, i will have to first find all the transactions ids. In that case retrieveTrns will again make an unnecessary call to dao. I think this can be a good way also.

retrieveTrns(
        dbCall = {
            transactionDao.findAllByAccountAndBetween(
                accountId = accountId.value,
                startDate = startDate,
                endDate = endDate
            )
        },
        retrieveTags = {
            getTagsForTransactionIds(listOf(it))[it.id] ?: emptyList()
        }
    )

Reply:
Hey @ILIYANGERMANOV , if you could confirm this, i can proceed